ethabi/src/decoder.rs

```diff
@@ -510,5 +539,68 @@ mod tests {
         assert!(decode(&[ParamType::FixedBytes(0)], &[]).is_ok());
         assert!(decode(&[ParamType::FixedArray(Box::new(ParamType::Bool), 0)], &[]).is_ok());
 	}
+
+    #[test]
+    fn decode_static_tuple_of_addresses_and_uints() {
+        let encoded = hex!(
+            "
+			0000000000000000000000001111111111111111111111111111111111111111
+			0000000000000000000000002222222222222222222222222222222222222222
+			1111111111111111111111111111111111111111111111111111111111111111
+		"
+        );
+        let address1 = Token::Address([0x11u8; 20].into());
+        let address2 = Token::Address([0x22u8; 20].into());
+        let uint = Token::Uint([0x11u8; 32].into());
+        let tuple = Token::Tuple(vec![address1, address2, uint]);
+        let expected = vec![tuple];
+        let decoded = decode(&[ParamType::Tuple(vec![Box::new(ParamType::Address),Box::new(ParamType::Address), Box::new(ParamType::Uint(32))])], &encoded).unwrap();
+        assert_eq!(decoded, expected);
+    }
+
+    #[test]
+    fn encode_dynamic_tuple() {
```

Comment:
I'd like to see a test where the tuple is inbetween other types rather than at the end. That would catch problems in advancing the `new_offset` in the tuple decoding.